### PR TITLE
Convert delete.test.ts to MSW

### PR DESCRIPTION
**Pull request by JacobMGEvans**
Continued efforts to convert to Mock Service Workers and remove custom mocking.

related tasks #1874

### CODE REVIEW

File: packages/wrangler/src/__tests__/delete.test.ts [hunk 1]
```diff
@@ -155,22 +157,40 @@ describe("delete", () => {
...
 		);
 
-		setMockResponse(
-			"/accounts/:accountId/storage/kv/namespaces/id-for-my-script-site-preview-ns",
-			"DELETE",
-			([_url, accountId]) => {
-				expect(accountId).toEqual("some-account-id");
-				return null;
-			}
+		msw.use(
```
Comment (penalosa) [inline packages/wrangler/src/__tests__/delete.test.ts:178]: This msw mock seems like it's used a lot, can it be extracted out into a function? Something like `mockDeleteSites()`?

File: packages/wrangler/src/__tests__/delete.test.ts [hunk 2]
```diff
@@ -197,34 +217,47 @@ function mockDeleteWorkerRequest(
...
-					: `${name || "test-name"}`
-			);
+	msw.use(
+		rest.delete(
+			"*/accounts/:accountId/workers/services/:scriptName",
+			(req, res, ctx) => {
+				expect(req.params.accountId).toEqual("some-account-id");
+				expect(req.params.scriptName).toEqual(
+					legacyEnv && env
+						? `${name || "test-name"}-${env}`
```
Comment (penalosa) [inline packages/wrangler/src/__tests__/delete.test.ts:227]:
```suggestion
						? `${name ?? "test-name"}-${env}`
```

File: packages/wrangler/src/__tests__/delete.test.ts [hunk 3]
```diff
@@ -197,34 +217,47 @@ function mockDeleteWorkerRequest(
...
+						: `${name || "test-name"}`
```
Comment (penalosa) [inline packages/wrangler/src/__tests__/delete.test.ts:228]: As above

File: packages/wrangler/src/__tests__/delete.test.ts (hunk 2 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/delete.test.ts:227]: This was previously existing, but I can make the change.

File: packages/wrangler/src/__tests__/delete.test.ts (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/delete.test.ts:178]: Rewrite is currently the focus. Cleanup and abstractions are welcome as follow up PRs

**Comment by rozenmd** [comment]
@JacobMGEvans can you make a follow-up issue to refactor the commonly used msw stuff? and @penalosa can you re-review

**Comment by penalosa** [comment]
It definitely feels like it would be simplest to make the refactoring changes as part of the rewrite (especially given the tails test, which took code that had common functionality extracted, and replaced it with boilerplate)?

**Comment by JacobMGEvans** [comment]
> @JacobMGEvans can you make a follow-up issue to refactor the commonly used msw stuff? and @penalosa can you re-review

Yeah I can add that issue.

**Review by penalosa — APPROVED** [review]

**Pull request merged**